core/src/main/java/org/apache/calcite/rel/rules/ProjectAggregateMergeRule.java

```diff
@@ -99,25 +100,27 @@ && kindCount(project.getProjects(), SqlKind.CASE) == 0) {
               && operands.get(1).getKind() == SqlKind.CAST
               && ((RexCall) operands.get(1)).operands.get(0).getKind()
               == SqlKind.INPUT_REF
-              && operands.get(2).getKind() == SqlKind.LITERAL) {
+              && operands.get(2).getKind() == SqlKind.LITERAL
+              && operands.get(2).getType().getFamily() == SqlTypeFamily.NUMERIC) {
```

Comment:
I was meant to remove this check originally since we can assume the operand is numeric when the aggregator is `SUM`. Sorry that this was still in my git staging.